### ISSUE TITLE
Issue with vendored libraries in Go 1.6

### ISSUE THREAD

**Issue by pavel-odintsov** [joefitzgerald/go-plus#445]
Hello!

I'm using Go 1.6 on Mac OS El Capitan with up-to-date go-plus and I have issues with Go's vendored folders.

I'm using following code for Atom startup:
```
export GOPATH=/Users/pavel/vm/myproject/.GOPATH
atom
```

Then I opened folder of my project under .GOPATH folder: /Users/pavel/vm/myproject/.GOPATH/src/code.mycompany.com/sys/dns/dns

But Atom show show a lot off errors about absent libraries which are located in vendor folder.

But I could manually build project with go build or go install from same folder:
```
cd /Users/pavel/vm/myproject/.GOPATH/src/code.mycompany.com/sys/dns/dns
go build
go install .
```

How I could do it with go-plus?

**Comment by joefitzgerald** [comment]
Please provide examples of the errors.

**Comment by pavel-odintsov** [comment]
I've getting this error: 
<img width="774" alt="screen shot 2016-07-08 at 15 18 32" src="https://cloud.githubusercontent.com/assets/2744166/16687135/60a45252-451f-11e6-9ac4-951409a6b9ea.png">

I've evaluated this issue in details and could share following:

- If I'm editing file from root project folder (which consists func Main) it working like a charm
- If I'm editing file which located in subfolder of main project I'm getting this error about vendored libraries.

So I think go-plus tried to build certain file instead of whole project....

**Comment by shadowmint** [comment]
I'm seeing this issue as well.

Go-plus version: 4.1.0

Go version: go version go1.6.2 windows/amd64

Oddly, it seems to work fine on my Mac, but the windows version consistently fails to locate packages in the vendor folder.

**Comment by joefitzgerald** [comment]
This is odd. Is there any way you can provide a reproduction in the form of a git repo I can clone to a fresh GOPATH?

**Comment by shadowmint** [comment]
While I was trying to push something up to github to show the issue I figured out what was going on here; the makefile was altering the GOPATH during the build, so that the project was sitting inside the GOPATH. 

Since the system level GOPATH is picked up by atom, and the `vendor` folder was *outside that path*, it was not detected by the build tool, and, not picked up by go-plus.

...and on my mac, it just so happened that it *was* inside the GOPATH, so it worked.

tldr; I don't think there's actually an issue here; just make absolutely sure that you project sits inside the GOPATH; vendor folders outside the GOPATH are ignored.

**Comment by pavel-odintsov** [comment]
@joefitzgerald I've tried hard to reproduce it but I could not do it with small project.

But I saw one interesting issue. When I trying to build file with main() function from top folder with fake "broken library" I got error where vendor folder mentioned explicitly: 
![error with vendor folder](https://cloud.githubusercontent.com/assets/2744166/16910505/1a85f0e6-4cea-11e6-8022-ffeeed34866b.jpg)

But when I tried to build some file which located in project's nested folders (mainapp/tools) I got error message without vendor folder:
<img width="908" alt="error without vendor folder" src="https://cloud.githubusercontent.com/assets/2744166/16910571/6f2b0e9c-4cea-11e6-86f7-b1c555b8be22.png">

And it's very strange. For some reasons "go install" could not handle vendor folder when I tried to build nested project's file.

Is there is any way how I could debug commands which Go Plus uses for building my project? Current working folder and explicit build command could help me to investigate this issue.

**Comment by pavel-odintsov** [comment]
I have great news, I reproduced it!

My project's root folder is: /Users/pavel/vm/rrdns/.GOPATH/src/code.my-company.com/sys/project/project

When I tried to build inside this folder everything working OK.

But when I tried to move to the nested folder: util (/Users/pavel/vm/rrdns/.GOPATH/src/code.my-company.com/sys/project/project/util) I still can build it.

But as I can understand Go Plus uses following way:
```
cd /Users/pavel/vm/rrdns/.GOPATH/src/code.my-company.com/sys/project/project/util
PWD=/Users/pavel/vm/rrdns/.GOPATH/src/code.cfops.it/sys/rrdns/rrdns go build
```

And I've got a lot of error about vendored folders without mentioned "vendor" path:
```
dpconn.go:4:2: cannot find package "golang.org/x/net/ipv4" in any of:
	/usr/local/go/src/golang.org/x/net/ipv4 (from $GOROOT)
	/Users/pavel/vm/rrdns/.GOPATH/src/golang.org/x/net/ipv4 (from $GOPATH)
```
 
I'm suppose bug inside logic of CWD/PWD override in packages/builder-go/lib/builder.js.

Here you could find debug output (with dump console.log) for executor params: 
![go plus issue jpeg](https://cloud.githubusercontent.com/assets/2744166/16912340/86b797ce-4cf4-11e6-8b79-05448fe4e083.jpg)

I've captured it from project which could not be builded due to issues with vendored folders.

I hope this could help you!

**Comment by pavel-odintsov** [comment]
Hello!

As I investigated this issue is not related with Go Plus at all. You could find more details: https://bugs.chromium.org/p/chromium/issues/detail?id=629437 

So we could close this issue.
